core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -2346,6 +2347,19 @@ void cancelAndClearAcquisitionLockTimeoutTask() {
             acquisitionLockTimeoutTask = null;
         }
 
+        private RecordState rollbackState() {
+            if (rollbackState == null) {
+                // This case could occur when the batch/offset hasn't transitioned even once or the state transitions have
+                // been committed.
+                return null;
+            }
+            return rollbackState.state;
+        }
```

Comment:
Do we need this method or should incorporate the change in `hasOngoingStateTransition`? Why to have this addiitonal logic which is confusing anyways in this method.